dc/s2n-quic-dc/src/stream/server/tokio/tcp/worker.rs

```diff
@@ -538,3 +566,255 @@ impl WorkerState {
         }
     }
 }
+
+#[derive(Clone)]
+pub struct SocketEventData {
+    blocked_count: usize,
+    payload_len: usize,
+    credential_id: Vec<u8>,
+    stream_id: u64,
+    sojourn_time: Duration,
+}
+
+#[derive(Clone)]
+pub struct SocketBehavior {
+    dest_path: PathBuf,
+}
+
+impl SocketBehavior {
+    #[inline]
+    pub fn new(dest_path: &Path) -> Self {
+        Self {
+            dest_path: dest_path.to_path_buf(),
+        }
+    }
+
+    fn poll_send<Pub, F>(
+        mut future: Pin<Box<F>>,
+        cx: &mut task::Context,
+        state: &mut WorkerState,
+        mut event_data: SocketEventData,
+        publisher: &Pub,
+    ) -> Poll<Result<ControlFlow<()>, Option<io::Error>>>
+    where
+        Pub: EndpointPublisher,
+        F: std::future::Future<Output = Result<(), std::io::Error>> + Send + 'static,
+    {
+        match future.as_mut().poll(cx) {
+            Poll::Ready(res) => match res {
+                Ok(_) => {
+                    publisher.on_acceptor_tcp_socket_sent(event::builder::AcceptorTcpSocketSent {
+                        credential_id: &event_data.credential_id,
+                        stream_id: event_data.stream_id,
+                        payload_len: event_data.payload_len,
+                        blocked_count: event_data.blocked_count,
+                        sojourn_time: event_data.sojourn_time,
```

Comment:
We either need a separate sojourn time or we should capture including the time it takes for us to reach Ready on the send. I would lean towards the later for now.